app/controllers/application_controller.rb

```diff
@@ -253,6 +254,15 @@ def needs_redirect?(host, domain)
     domain.present? && host != domain
   end
 
+  def redirect_to_marketplace_ident
+    host = request.host
+    app_domain = URLUtils.strip_port_from_host(APP_CONFIG.domain)
+    if host.start_with?("www.") && host.end_with?(app_domain)
+      ident_without_www = host.sub(/www\./, '').chomp(".#{app_domain}")
+      return redirect_to "#{request.protocol}#{ident_without_www}.#{APP_CONFIG.domain}"
```

Comment:
[Google recommends](https://support.google.com/webmasters/answer/139066?hl=en#4) using a 301 redirect for non-canonical urls, `redirect_to` uses 302 as default: http://guides.rubyonrails.org/layouts_and_rendering.html#using-redirect-to